core/network/src/main/java/com/mateuszholik/network/repositories/MatchesApiRepository.kt

```diff
@@ -0,0 +1,52 @@
+package com.mateuszholik.network.repositories
+
+import com.mateuszholik.network.extensions.asString
+import com.mateuszholik.network.extensions.toResultApi
+import com.mateuszholik.network.models.Head2HeadApi
+import com.mateuszholik.network.models.MatchApi
+import com.mateuszholik.network.models.MatchesApi
+import com.mateuszholik.network.models.ResultApi
+import com.mateuszholik.network.services.MatchesService
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOn
+import java.time.LocalDate
+
+interface MatchesApiRepository {
+
+    fun getMatchesForDateRange(dateFrom: LocalDate, dateTo: LocalDate): Flow<ResultApi<MatchesApi>>
+
+    fun getMatch(id: Int): Flow<ResultApi<MatchApi>>
+
+    fun getHead2HeadForMatch(id: Int): Flow<ResultApi<Head2HeadApi>>
+}
+
+internal class MatchesApiRepositoryImpl(
+    private val matchesService: MatchesService,
+    private val ioDispatcher: CoroutineDispatcher,
+) : MatchesApiRepository {
+
+    override fun getMatchesForDateRange(
+        dateFrom: LocalDate,
+        dateTo: LocalDate,
+    ): Flow<ResultApi<MatchesApi>> =
```

Comment:
Why do you want to have flow here? I think it's better to have simple suspend fun